www/public/index.html

```diff
@@ -41,8 +55,241 @@ <h1>Work in Progress</h1>
+
+<p>Work has not yet started on the package manager, static analyzer, or debugger, and although work
+  has started on the editor, it's not yet far enough along to be usable for practical purposes. The
+  standard library is perhaps 80 percent complete in terms of functionality, but a lot of operations
+  do not yet have documentation.</p>
+
+<p>The REPL fully supports entering arbitrary expressions, and will evaluate them and print the
+  results. It remembers recent expressions entered in the current session (if you press the up arrow),
+  but it can't yet execute effects, and any named variables you define currently exist only for the
+  next expression it evaluates; they don't yet persist for the remainder of the session. You can try
+  out the REPL in a browser at <a href="roc-lang.org/repl">roc-lang.org/repl</a> - it uses a
+  WebAssembly build of Roc's compiler, and compiles the code you write to WebAssembly on the fly,
+  which it then executes in the browser to display the answer.</p>
+
+<p>The compiler works well enough on a basic level to build things with it, but some error messages
+  could use significant improvement, and it has a lot of known bugs and missing features. You can
+  currently use it on macOS (either Intel or Apple Silicon), Linux (only x86 machines at the moment),
+  and Windows (only recently supported; there are definitely still a number of issues to iron out).
+  Support for other operating systems has not yet been discussed.</p>
+
+<p>In the future, the entire Roc toolset should work completely as a standalone binary, but for now,
+  application development requires additionally having some combination of
+  <a href="https://clang.llvm.org">clang</a>,
+  <a href="https://www.rust-lang.org/">Rust</a>, and
+  <a href="https://ziglang.org">Zig</a> installed on your machine. These are needed to compile
+  certain platform dependencies on the fly; in the future, the plan is to have the package manager
+  include precompiled target-specific binaries for these dependencies, so the only binary you'll need
+  to have is the one that includes the Roc compiler itself.</p>
+
+<p>The compiler doesn't yet support incremental compilation or hot code loading, and build times vary
+  based on what machine you're building for.</p>
+
+<p>For example, suppose you run `roc check`, which reports errors it finds (type mismatches, naming
+  errors, and so on) but doesn't actually build an executable, on a code base that's under a thousand
+  lines of code. On an M1 MacBook Pro, this typically takes about 10 milliseconds.</p>
+
+<p>In contrast, if you do `roc build` (or `roc run`) on that same machine, it will take closer to 500
+  milliseconds instead. Almost all that extra time is spent waiting for LLVM to generate (unoptimized)
+  machine code, and then for the system linker to assemble an executable from it. This is because on
+  macOS we don't yet have a development backend or surgical linking set up yet.</p>
```

Comment:
Most people probably will not be familiar with a "development backend" or "surgical linking" :)